orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/model/Stage.java

```diff
@@ -357,13 +357,30 @@ public Task taskById(String taskId) {
         .addAll(previousStages.stream().flatMap(it -> it.ancestorsOnly().stream()).collect(toList()))
         .build();
     } else if (parentStageId != null) {
-      return execution.getStages().stream().filter(it -> it.id.equals(parentStageId)).findFirst()
-        .<List<Stage>>map(parent -> ImmutableList
-          .<Stage>builder()
-          .add(parent)
-          .addAll(parent.ancestorsOnly())
-          .build())
-        .orElse(emptyList());
+      List<Stage> ancestors = new ArrayList<>();
+      if (getSyntheticStageOwner() == SyntheticStageOwner.STAGE_AFTER) {
```

Comment:
Not sure if there was a good reason to not consider sibling stages when looking for ancestors.

The particular issue in question arises because a `STAGE_BEFORE` produces an output that a `STAGE_AFTER` expects to read.